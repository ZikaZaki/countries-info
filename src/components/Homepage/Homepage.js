import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchCountries } from '../../redux/configureStore';
import CountryList from '../CountryList/CountryList';
import world from '../../assets/images/World.svg';

// Import Styling
import styles from './Homepage.module.css';

const Homepage = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries);
  const [filter, setFilter] = useState('');
  const filterCountries = (e) => {
    if (e.target.value === 'All') {
      setFilter(countries);
    } else {
      setFilter(countries.filter((country) => country.continent === e.target.value));
    }
  };

  useEffect(() => {
    dispatch(fetchCountries());
  }, [dispatch]);

  return (
    <div className={styles['home-container']}>
      <div className={styles['world-map']}>
        <div className={styles.map}>
          <img src={world} alt="World Map" />
        </div>
        <div className={styles['map-info']}>
          <h2>WORLD MAP</h2>
          <p>
            8,000
            <br />
            countries
          </p>
        </div>
      </div>
      <div className={styles.headline}>
        <h2>STATS BY CONTINENTS</h2>
        <label htmlFor="continents">
          <select
            type="text"
            value="Select Continent"
            onChange={() => console.log('clicked')}
            placeholder="Select Your Continent"
            name="continent"
            className="button"
          >
            <option value="All">All</option>
            <option value="Asia">Asia</option>
            <option value="Africa">Africa</option>
            <option value="Antarctica">Antarctica</option>
            <option value="Europe">Europe</option>
            <option value="Oceania">Oceania</option>
            <option value="North America">North America</option>
            <option value="South America">South America</option>
          </select>
        </label>
      </div>
      <CountryList countries={countries} />
      {/* <CountryList countries={filter || countries} /> */}
    </div>
  );
};

export default Homepage;
