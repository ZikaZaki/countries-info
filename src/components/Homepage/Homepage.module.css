.home-container {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  font-family: var(--lato);
  min-height: 100vh;
  width: 100%;

  /* background-color: var(--primary); */
  background-color: #ff5294;
  color: var(--neutral);
}

.world-map {
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  gap: 0.5rem;
  height: 40vh;
  width: 100%;
  padding: 1rem;
}

.map {
  display: flex;
  height: 100%;
  width: 100%;
  padding: 0;
}

.map img {
  display: flex;
  width: 100%;
  height: 100%;
  object-fit: contain;
}

.map-info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 0.5rem;
  height: auto;
  width: auto;
  padding: 0;
  position: absolute;
  top: 25%;
  right: 20%;
  transform: translate(50%, -50%);
}

.map-info h2 {
  font-size: calc(10px + 1vmin);
  font-weight: bold;
  text-align: center;
  padding: 0;
  margin: 0;
}

.map-info p {
  text-align: center;
  font-weight: 300;
  font-size: 1rem;
  padding: 0;
  margin: 0;
}

.headline {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  vertical-align: middle;
  gap: 0.5rem;
  height: 5vh;
  width: 100%;
  padding: 0.2rem 1rem;
  background-color: var(--primary);
}

.headline h2 {
  font-size: calc(8px + 1vmin);
  font-weight: bold;
  text-align: center;
  padding: 0;
  margin: 0;
}

.headline select {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: calc(8px + 1vmin);
  background-color: #a53561;
  text-align: justify;
  color: var(--neutral);
  border-radius: 0.3rem;
  border: #a53561 1px solid;
  padding: 0;
  margin: 0;
}
