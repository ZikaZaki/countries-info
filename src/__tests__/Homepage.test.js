import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Homepage from '../components/Homepage/Homepage';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Homepage />
        ,
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the component container is there', async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('home-container');
  expect(container).toBeInTheDocument();
});
