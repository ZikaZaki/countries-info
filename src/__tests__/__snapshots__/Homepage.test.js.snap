// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
Array [
  <div
    className="home-container"
    data-testid="home-container"
  >
    <div
      className="world-map"
    >
      <div
        className="map"
      >
        <img
          alt="World Map"
          src="World.svg"
        />
      </div>
      <div
        className="map-info"
      >
        <h2>
          World Map
        </h2>
        <p>
          2022
        </p>
      </div>
    </div>
    <div
      className="headline"
    >
      <h2>
        STATS BY CONTINENTS
      </h2>
      <label
        htmlFor="continents"
      >
        <select
          className="button"
          name="continent"
          onChange={[Function]}
          placeholder="Select Your Continent"
          type="text"
          value="Select Continent"
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="Asia"
          >
            Asia
          </option>
          <option
            value="Africa"
          >
            Africa
          </option>
          <option
            value="Americas"
          >
            Americas
          </option>
          <option
            value="Europe"
          >
            Europe
          </option>
          <option
            value="Oceania"
          >
            Oceania
          </option>
        </select>
      </label>
    </div>
    <div
      className="country-list"
      data-testid="countryList-container"
    />
  </div>,
  ",",
]
`;
